### PR TITLE
Update Detekt 1.4.0 → 1.5.1.

### PR DESCRIPTION
**Pull request by arturdryomov**
[The changelog](https://github.com/arturbosch/detekt/releases).

* Good news: seems like our OOM fix motivated the team to fix even more leaks.
* Bad news: Detekt is strict by default now, so I had to refactor integration tests to address this.

### CODE REVIEW

**Comment by Kernald** [comment]
Also, Detekt 1.7.0 has been released, but it probably makes sense finishing work on integrating this 1.5.1 and then move on to the next one :-)

File: tests/integration/suite.sh [hunk 1]
```diff
@@ -5,15 +5,13 @@ GENERATED_CODE_DIR="tests/integration/src/main/kotlin/generated"
 
 for STRATEGY in "local" "worker"; do
     echo ":: Executing with the [${STRATEGY}] strategy."
-    echo "build --strategy=Detekt=${STRATEGY}" > ".bazelrc"
+    echo "build --strategy=Detekt=${STRATEGY}" >> ".bazelrc"
```
[the remark this answers is not in the document]
Comment (arturdryomov) [reply tests/integration/suite.sh]: That’s a great question, I’m not comfortable with this statement as well. We have a custom `.bazelrc` on CI to change cache locations with the goal to persist the cache between runs. At the same time, this suite script executes all tests using two strategies. A better idea would be backing up the `.bazelrc` between strategy changes, I’ll check.

**Comment by artem-zinnatullin** [comment]
Let's merge this @arturdryomov? I want to update it in our project :)

**Comment by arturdryomov** [comment]
On it! Sorry for the timeout, too much code in life while WFH.

**Pull request merged**